src/energy_systems/heat_sources/generic_heat_source.jl

```diff
@@ -0,0 +1,197 @@
+            ),
+            sf_bounded_source, # sys_function
+            medium, # medium
+            InterfaceMap( # input_interfaces
+                medium => nothing
+            ),
+            InterfaceMap( # output_interfaces
+                medium => nothing
+            ),
+            max_power_profile,
+            temperature_profile,
+            default(config, "scale", 1.0), # scaling_factor
+            default(config, "constant_power", nothing),
+            default(config, "constant_temperature", nothing),
+            default(config, "use_exchange_loss", false),
+            default(config, "min_source_in_temperature", nothing),
+            default(config, "max_source_in_temperature", nothing),
+            default(config, "delta_loss_lower", 2.5),
+            default(config, "delta_loss_upper", 1.0),
+            0.0, # max_energy
+            nothing, # temperature_src_in
+            nothing, # temperature_snk_out
+        )
+    end
+end
+
+function initialise!(unit::GenericHeatSource, sim_params::Dict{String,Any})
+    set_storage_transfer!(
+        unit.output_interfaces[unit.medium],
+        default(
+            unit.controller.parameter, "load_storages " * String(unit.medium), true
+        )
+    )
+
+    if (
+        unit.use_exchange_loss
+        && unit.temperature_profile !== nothing
+    )
+        if unit.min_source_in_temperature === nothing
+            unit.min_source_in_temperature = Profiles.minimum(unit.temperature_profile)
+        end
+        if unit.max_source_in_temperature === nothing
+            unit.max_source_in_temperature = Profiles.maximum(unit.temperature_profile)
+        end
+    end
+end
+
+function control(
+    unit::GenericHeatSource,
+    components::Grouping,
+    sim_params::Dict{String,Any}
+)
+    move_state(unit, components, sim_params)
+
+    if unit.constant_power !== nothing
+        unit.max_energy = watt_to_wh(unit.constant_power)
+    elseif unit.max_power_profile !== nothing
+        unit.max_energy = unit.scaling_factor * Profiles.work_at_time(
+            unit.max_power_profile, sim_params["time"]
+        )
+    else
+        unit.max_energy = 0.0
+    end
+    set_max_energy!(unit.output_interfaces[unit.medium], unit.max_energy)
+
+    if unit.constant_temperature !== nothing
+        unit.temperature_src_in = unit.constant_temperature
+    elseif unit.temperature_profile !== nothing
+        unit.temperature_src_in = Profiles.value_at_time(
+            unit.temperature_profile, sim_params["time"]
+        )
+    end
+
+    if unit.use_exchange_loss && unit.temperature_src_in !== nothing
+        unit.temperature_snk_out = unit.temperature_src_in -
+            (unit.temperature_src_in - unit.min_source_in_temperature) /
+            (unit.max_source_in_temperature - unit.min_source_in_temperature) *
+            (unit.delta_loss_lower - unit.delta_loss_upper) -
+            unit.delta_loss_upper
```

Comment:
I'm not sure about this model. Perhaps a smarter way would be to use the distance from the source input temperature from the "design temperature" at which the heat exchanger is optimal. Let's discuss this.